packages/api/tests/lib/rules/test-createRuleResources.js

```diff
@@ -0,0 +1,358 @@
+  const logEventMappings = kinesisEventMappings[1].EventSourceMappings;
+
+  t.is(consumerEventMappings.length, 1);
+  t.is(logEventMappings.length, 1);
+  t.is(consumerEventMappings[0].UUID, createdRule.arn);
+  t.is(logEventMappings[0].UUID, createdRule.log_event_arn);
+
+  t.is(createdRule.name, kinesisRule.name);
+  t.is(createdRule.value, kinesisRule.value);
+  t.false(createdRule.arn === undefined);
+  t.false(createdRule.log_event_arn === undefined);
+
+  // Clean Up
+  // await deleteRuleResources(t.context.testKnex, kinesisRule);
+  await deleteKinesisEventSourceMappings();
+});
+
+test('Creating an invalid kinesis type rule does not add event mappings', async (t) => {
+  const kinesisRule = fakeRuleRecordFactory({
+    type: 'kinesis',
+    workflow,
+    enabled: true,
+    value: randomString(),
+  });
+  delete kinesisRule.name;
+  await t.throwsAsync(createRuleTrigger(kinesisRule), { name: 'SchemaValidationError' });
+
+  const kinesisEventMappings = await getKinesisEventMappings();
+  const consumerEventMappings = kinesisEventMappings[0].EventSourceMappings;
+  const logEventMappings = kinesisEventMappings[1].EventSourceMappings;
+
+  console.log(JSON.stringify(kinesisEventMappings));
+
+  t.is(consumerEventMappings.length, 0);
+  t.is(logEventMappings.length, 0);
+});
+
+test('Creating a rule trigger SQS rule fails if queue does not exist', async (t) => {
+  const rule = fakeRuleRecordFactory({
+    workflow,
+    type: 'sqs',
+    value: 'non-existent-queue',
+    enabled: true,
+  });
+  await t.throwsAsync(
+    createRuleTrigger(rule),
+    { message: /SQS queue non-existent-queue does not exist/ }
+  );
+});
+
+test('Creating a rule trigger for an SQS rule fails if there is no redrive policy on the queue', async (t) => {
+  const queueUrl = await SQS.createQueue(randomId('queue'));
+  const rule = fakeRuleRecordFactory({
+    workflow,
+    type: 'sqs',
+    value: queueUrl,
+    enabled: true,
+  });
+  await t.throwsAsync(
+    createRuleTrigger(rule),
+    { message: `SQS queue ${queueUrl} does not have a dead-letter queue configured` }
+  );
+  t.teardown(async () => await SQS.deleteQueue(queueUrl));
+});
+
+test('Creating a rule trigger for an SQS rule succeeds', async (t) => {
+  const queues = await createSqsQueues(randomString());
+  const rule = fakeRuleRecordFactory({
+    workflow,
+    type: 'sqs',
+    value: queues.queueUrl,
+    enabled: true,
+    meta: {
+      visibilityTimeout: 100,
+      retries: 4,
+    },
+  });
+  const sqsRule = await createRuleTrigger(rule);
+  t.deepEqual(sqsRule, rule);
+  t.teardown(async () => await SQS.deleteQueue(queues.queueUrl));
+});
+
+test('Creating a rule trigger for an SQS rule succeeds sets default value for meta.retries and visibilityTimeout', async (t) => {
+  const queues = await createSqsQueues(randomString());
+  const rule = fakeRuleRecordFactory({
+    workflow,
+    type: 'sqs',
+    value: queues.queueUrl,
+    enabled: true,
+    meta: {},
```

Comment:
This shouldn't be necessary, I believe

Reply:
[`907d615` (#2871)](https://github.com/nasa/cumulus/pull/2871/commits/907d6158686dd2f2b5bc640e15c153114fce40ae#diff-f5c5b96c6230da09644104acbdca2268e22d32509bc5b275a5b4648a8939584eL236)